### PR TITLE
Misc linalg tests yet again

### PR DESCRIPTION
**Pull request by kshyatt**
`scale!` for the `AbstractTriangular`s was still missing a test or two.

`generic.jl` had a few error branches and methods untested.

### CODE REVIEW

File: test/linalg/generic.jl [hunk 1]
```diff
@@ -56,10 +56,17 @@ y = ['a','b','c','d','e']
...
 @test_throws BoundsError Base.LinAlg.axpy!(α,x,collect(-1:5),y,collect(1:7))
+@test_throws BoundsError Base.LinAlg.axpy!(α,x,collect(1:7),y,collect(-1:5))
+@test_throws ArgumentError Base.LinAlg.axpy!(α,x,collect(1:3),y,collect(1:5))
 @test_throws BoundsError Base.LinAlg.axpy!(α,x,collect(1:7),y,collect(1:7))
+@test_throws DimensionMismatch Base.LinAlg.axpy!(α,x,collect(1:2),['a','b'],collect(1:2))
 
+@test_throws ArgumentError diag(rand(10))
+@test !issym(ones(5,3))
+@test !ishermitian(ones(5,3))
+@test cross(ones(3),ones(3)) ≈ zeros(3)
```
Comment (tkelman) [inline test/linalg/generic.jl]: should be exact?

Comment (kshyatt) [inline test/linalg/generic.jl]: Yep! Will update & resubmit

**Comment by tkelman** [comment]
Yeah this looks good. Pure test addition so should be fine to backport too.

**Comment by tkelman** [comment]
backported in #13107

**Pull request merged**